fleece/parameters.py

```diff
@@ -0,0 +1,130 @@
+    return param_list
+
+
+def _unflatten_config(param_list):
+    """Reconstructs a config dictionary from a list of key/value tuples."""
+    config = {}
+
+    def parse_key(key):
+        k = key
+        i = -1
+        if '.' in key:
+            k, i = key.split('.')
+            i = int(i)
+        return k, i
+
+    for param in param_list:
+        keys = param[0].split('/')
+
+        c = config
+        for k in keys[1:-1]:
+            key, index = parse_key(k)
+            if key not in c:
+                if index == -1:
+                    c[key] = {}
+                else:
+                    c[key] = []
+            c = c[key]
+            if index != -1:
+                while (len(c) < index + 1):
+                    c.append(None)
+                if c[index] is None:
+                    c[index] = {}
+                c = c[index]
+        key, index = parse_key(keys[-1])
+        if index == -1:
+            c[key] = param[1]
+        else:
+            if key not in c:
+                c[key] = []
+            c = c[key]
+            while (len(c) < index + 1):
+                c.append(None)
+            c[index] = param[1]
+
+    return config
+
+
+def read_parameters(root_path, **boto_args):
+    """Read parameters from EC2 Parameter Store previously written with the
+    write_parameters() function and return them as a configuration dictionary.
+    """
+    ssm = boto3.client('ssm', **boto_args)
+    r = ssm.describe_parameters(ParameterFilters=[{
+        'Key': 'Path',
+        'Option': 'Recursive',
+        'Values': [root_path]
+    }])
+    p = ssm.get_parameters(
+        Names=[param['Name'] for param in r['Parameters']],
+        WithDecryption=True)
+    param_list = [(param['Name'], param['Value']) for param in p['Parameters']]
+    return _unflatten_config(param_list)[root_path[1:]]
+
+
+def erase_parameters(root_path, **boto_args):
+    """Erase all parameters under the given path in the EC2 Parameter Store."""
+    ssm = boto3.client('ssm', **boto_args)
+    r = ssm.describe_parameters(ParameterFilters=[{
+        'Key': 'Path',
+        'Option': 'Recursive',
+        'Values': [root_path]
+    }])
+    for param in r['Parameters']:
+        ssm.delete_parameter(Name=param['Name'])
+
+
+def write_parameters(config, kms_key, root_path, **boto_args):
+    """Write a config file dictionary to the EC2 Parameter Store."""
+    param_list = _flatten_config(config)
+    ssm = boto3.client('ssm', **boto_args)
+
+    # first remove old parameters
+    r = ssm.describe_parameters(ParameterFilters=[{
+        'Key': 'Path',
+        'Option': 'Recursive',
+        'Values': [root_path]
+    }])
+    for param in r['Parameters']:
+        ssm.delete_parameter(Name=param['Name'])
+
+    # write new parameters
+    for param in param_list:
+        if param[2]:
+            ssm.put_parameter(Name=root_path + param[0], Value=param[1],
+                              Type='SecureString', KeyId='alias/' + kms_key)
```

Comment:
If we choose to use `KeyId` here instead of the default SSM key, I believe we will need to ensure that all the KMS keys' key policies allow any Lambda calling `get_parameters_by_path` to decrypt, which could be complicated. In light of this, I think it might be best to just use the default SSM key here and control access via to the parameters themselves via IAM.